src/chapters/http/how-the-internet-works.rst

```diff
@@ -0,0 +1,193 @@
+place to start. Now we're ready to take a deeper look at how the internet
+works.
+
+Most people use the internet without fully understanding what's going on behind
+the scenes. It's fairly easy to open a browser, navigate to a site, and
+interact with it. Knowing exactly *how* the internet works isn't really
+necessary for most folks.
+
+However, for web developers like us, understanding how information flows across
+the internet is essential.
+
+Servers and Clients
+-------------------
+
+.. index:: ! client-server model, ! client, server
+
+The Internet uses the **client-server model**. A **server** is a machine that
+provides resources to other devices. These resources include raw data, web
+pages, images, etc. A **client** is an application or device that *requests*
+resources from a server.
+
+.. admonition:: Example
+
+   Let's take a look at one type of client-server exchange. Imagine we are
+   surfing the web on our computer or smartphone.
+
+   #. The *client* is the web browser on our device.
+   #. When we click on a link or type in an address and hit *Enter*, the client
+      sends a request to a server.
+   #. This *server* (which sits in a building somewhere out in the world)
+      receives our request and sends back a response.
+   #. The response contains some amount of data, which is written in HTML code.
+   #. The client receives the response, executes the code, and displays the
+      data in a way we can understand.
+
+.. index:: ! internet service provider
+
+Note that clients, like our smartphones and laptops, are not connected directly
+to the internet. Instead, they interact with an **internet service provider**
+(ISP), which provides a link between clients and the internet. Examples of ISPs
+include Charter, Verizon, and AT&T.
+
+To update our picture, let's connect some extra equipment to the wire.
+
+.. figure:: figures/client-server.png
+   :alt: A server connected by wires to two client computers. Arrows indicate a request/response.
+   :width: 70%
+
+   Two clients communicating with a server.
+
+Addresses
+---------
+
+When we make a request inside our web browser, how does the client know where
+to send it? Also, once it receives a request, how does the server know where to
+send the data?
+
+.. index:: ! IP address
+
+Every device connected directly or indirectly to the internet has its own
+numerical address, called an **IP address**. "IP" stands for *internet
+protocol*. The links we click on a webpage contain IP addresses, which identify
+the server we need to contact. Requests also include the address of our device.
+
+Any time our request passes through a different device on its way to the server,
+that device adds its own IP address to the request. Think of this like a trail
+of breadcrumbs, or as a series of directions on a map. When the server responds
+to our request, it sends the data through the same IP addresses, but in reverse
+order.
+
+.. figure:: figures/ip-trail.png
```

Comment:
I like this diagram. However (now this is a pedantic point, I know) the IP address examples wouldn't occur in the real world. IPs like 192.168.x.x are always _local_ IPs within a local network, like your home. So routers directing traffic on the internet would not have these IPs. If you made the two routers have IPs more like the server (68.63.x.x) and vice versa, that would be a real world example.

Reply:
OK.  I changed the IP addresses on the routers and server.